src/linearmixedmodel.jl

```diff
@@ -78,6 +78,18 @@ function LinearMixedModel(
     for (i, x) in enumerate(Xs)
         if isa(x, AbstractReMat{T})
             push!(reterms, x)
+        elseif isa(x, ReMat) # this can occur in weird situation where x is a ReMat{U}
+            # avoid keeping a second copy if unweighted
+            z = convert(Matrix{T}, x.z)
```

Comment:
I tend to write this construction as `T.(x.z)` but this is quite reasonable too.